### DISCUSSION TITLE
Change sets Flow clarifications

### DISCUSSION THREAD

**Discussion by orenmizr** [changesets/changesets#1042] [Q&A]
please help me understand this tool in regards with monorepo (turborepo)

1. why doesn't changesets see which package in the monorepo i have worked on in the commit ? why does it ask the developer to choose ? 

2. how can i allow the developers to commit without writing the changeset and only describe the changes when they are ready ? when working most developer commit small changes not a the big one. how can that work ?

**Comment by Andarist** [comment]
> why doesn't changesets see which package in the monorepo i have worked on in the commit ?

We split the list of packages into changes and not changed packages - so we do recognize which packages you have worked on.

> why does it ask the developer to choose ?

Conceptually there is no 1-to-1 relationship between your changes and changesets. You are free to add more than one changeset per package and you are free to create different changesets for different packages for the very same "change". Those are not coupled. For each package that has been affected by your changes you probably should write a changelog entry from **its own** perspective.

> how can i allow the developers to commit without writing the changeset and only describe the changes when they are ready ? when working most developer commit small changes not a the big one. how can that work ?

Perhaps you are looking for a `changeset status` check for your PRs - you don't have to create changesets immediately when working on your changesets but it's advised to create them some time before you land the PR. A CI check could help you enforce this.

**Reply by orenmizr** [reply]
first, i should have started here. apologies on the first exchange. i must confess i have read the docs site twice and ran all the commands on different flows just to see how it would react. this is not a case of someone just asking without having to read the manual. it sounds that something in the setup work did not work with my turborepo otherwise the outputs would have been as the site states which corelates to your answers. i am a bit confused. do you have a repo for me to clone and play with. i have my own which sound broken:

1. i have created a turborepo default repo from the generator. than proceeded to changeset init. 
2. i have made changes to a package (acme) but it does not see it when running changeset add

(might be important: i do use pnpm)

**Reply by orenmizr** [reply]
when i ran changeset init on a cloned repo - i noticed it created the wrong base branch. (master vs main). then i looked in my actual project and noticed there was missing a base branch (also changes inited) which might explain the issues. checking now the flows.

**Reply by Andarist** [reply]
I rarely use `changeset init` myself - perhaps the values that are created by it could be improved. A configured base branch is most likely very important for the "change" calculation.

> i have made changes to a package (acme) but it does not see it when running changeset add

Maybe you didn't commit the changes yet? I think that we detect change purely based on commits - most likely we should also take the staging (and working area?) into the account there.

>  do you have a repo for me to clone and play with.

Well, in general - most of this stuff is not really coupled with the package manager of choice, so you should be able to take a look at just any repo that uses Changesets (including this repo)
